src/cairoUtilFuncGen/inputArgCheck/inputCheck.ts

```diff
@@ -0,0 +1,267 @@
+      code: [
+        `func ${funcName}${implicits}(arg : felt) -> ():`,
+        `    let (inRange : felt) = is_le_felt(arg, ${nMembers - 1})`,
+        `    with_attr error_message("Error: value out-of-bounds. Values passed to must be in enum range (0, ${
+          nMembers - 1
+        }]."):`,
+        `        assert 1 = inRange`,
+        `    end`,
+        `    return ()`,
+        `end`,
+      ].join('\n'),
+    });
+    this.requireImport('starkware.cairo.common.math_cmp', 'is_le_felt');
+    return funcName;
+  }
+
+  private createDynArrayInputCheck(key: string, type: ArrayType): string {
+    const funcName = `extern_input_check${this.generatedFunctions.size}`;
+    const implicits = '{range_check_ptr : felt}';
+
+    // Set an empty entry so recursive function generation doesn't clash
+    this.generatedFunctions.set(key, { name: funcName, code: '' });
+
+    const cairoType = CairoType.fromSol(type, this.ast, TypeConversionContext.CallDataRef);
+    assert(cairoType instanceof CairoDynArray);
+    const ptrType = cairoType.vPtr;
+    const elementType = generalizeType(type.elementT)[0];
+    this.checkForImport(elementType);
+    const cairoElmType = CairoType.fromSol(
+      elementType,
+      this.ast,
+      TypeConversionContext.CallDataRef,
+    );
+    const indexCheck = [`${this.getOrCreate(elementType)}(ptr[0])`];
+
+    this.generatedFunctions.set(key, {
+      name: funcName,
+      code: [
+        `func ${funcName}${implicits}(len: felt, ptr : ${ptrType.toString()}) -> ():`,
+        `    alloc_locals`,
+        `    if len == 0:`,
+        `        return ()`,
+        `    end`,
+        ...indexCheck,
+        `   ${funcName}(len = len - 1, ptr = ptr + ${
+          ptrType.to instanceof CairoFelt ? '1' : cairoElmType.toString() + '.SIZE'
+        })`,
+        `    return ()`,
+        `end`,
+      ].join('\n'),
+    });
+    return funcName;
+  }
+
+  checkableType(type: TypeNode): boolean {
+    return (
+      type instanceof ArrayType ||
+      (type instanceof UserDefinedType &&
+        (type.definition instanceof StructDefinition ||
+          type.definition instanceof EnumDefinition)) ||
+      type instanceof AddressType ||
+      type instanceof IntType ||
+      type instanceof BoolType
+    );
+  }
+
+  checkForImport(type: TypeNode): void {
+    if (
+      type instanceof UserDefinedType &&
+      (type.definition instanceof StructDefinition || type.definition instanceof EnumDefinition)
+    ) {
+      assert(this.sourceUnit !== undefined);
+      const typeDefSourceUnit = type.definition.root;
+      assert(typeDefSourceUnit instanceof SourceUnit);
+      if (this.sourceUnit !== typeDefSourceUnit) {
+        this.requireImport(this.formatPath(typeDefSourceUnit.absolutePath), type.definition.name);
+      }
+    } else if (type instanceof IntType && type.nBits === 256) {
+      this.requireImport('starkware.cairo.common.uint256', 'Uint256');
+    }
+  }
+
+  formatPath(path: string): string {
```

Comment:
Can this be exported/imported rather than having the same function here and in externImporter